src/tools/crawler/main.rs

```diff
@@ -94,29 +84,25 @@ async fn main() {
     // Create the crawler with the given listener address.
     let crawler = Crawler::new().await;
 
-    if let Some(seeds) = args.seed_addrs {
-        seed_addrs.extend_from_slice(&seeds);
-    }
-
-    if let Some(dns_seed) = args.dns_seed {
-        for seed_str in dns_seed {
-            let response = lookup_host(&seed_str);
+    for seed_addr in args.seed_addrs {
+        // First, try parsing as a `SocketAddr`.
+        if let Ok(addr) = seed_addr.parse::<SocketAddr>() {
+            seed_addrs.push(addr);
+        } else {
+            // If above failed, try to do a DNS lookup instead.
+            let response = lookup_host(&seed_addr);
 
             if let Ok(response) = response {
                 for address in response.iter() {
                     seed_addrs.push(SocketAddr::new(*address, ZCASH_P2P_DEFAULT_PORT)); // DNS addrs use this port
-                    info!(parent: crawler.node().span(), "DNS seed {} address added: {}", seed_str, address);
+                    info!(parent: crawler.node().span(), "DNS seed {} address added: {}", seed_addr, address);
                 }
             } else {
-                error!(parent: crawler.node().span(), "DNS seed {} lookup failed: {}", seed_str, response.err().unwrap());
+                error!(parent: crawler.node().span(), "DNS seed {} lookup failed: {}", seed_addr, response.err().unwrap());
```

Comment:
now well, this could be a general text simply saying "invalid {seed_addr}"

because one might fail to write an address e.g. `1288.100.13.4:8233` and we will print
`DNS seed 1288.100.13.4:8233 lookup failed`
which would be odd, no?